File: javascript solutions/singleNumber.js
// Given a non-empty array of integers, every element appears twice except for one. Find that single one.

// Note:

// Your algorithm should have a linear runtime complexity. Could you implement it without using extra memory?

// Example 1:
// Input: [2,2,1]
// Output: 1

// Example 2:
// Input: [4,1,2,1,2]
// Output: 4

function singleNumber(nums){
  let counter = {};
  nums.forEach((num) => {
    if (!counter[num]) {
      counter[num] = 1;
    } else {
      counter[num] += 1;
    }
  })
  for (const num in counter) {
    if (counter[num] === 1) {
      return num;
    }
  }
}

File: python solutions/single_number.py
# Given a non-empty array of integers, every element appears twice except for one. Find that single one.

# Note:

# Your algorithm should have a linear runtime complexity. Could you implement it without using extra memory?

# Example 1:

# Input: [2,2,1]
# Output: 1

from collections import Counter
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        counter = Counter(nums)
        for key in counter:
            if counter[key] == 1:
                return key